test/paths.ts

```diff
@@ -186,3 +186,27 @@ function testDepth2() {
     Assert<Assignable<P, "nested.tuple.2.good">>,
   ];
 }
+
+/**
+ * Configuration allows wildcard accessors for array indices.
+ */
+function testAnyArrayIndexAccessor() {
+  type With = Paths<ComplexNestedRequired, { anyArrayIndexAccessor: "*" }>;
+  type Without = Paths<ComplexNestedRequired>;
+
+  type cases_with_any_accessor = [
+    Assert<Has<With, "nested.array.*">>,
+    Assert<Has<With, "nested.array.*.bar">>,
+    Assert<Has<With, `nested.array.${number}`>>,
```

Comment:
During my testing, I found that `"nested.array.*"` will be acceptable, but not `` `nested.array.${number}` ``. I couldn't figure it out why this happens, so if you face it during your testing and cannot figure it out how to fix it, please leave it. It's a known bug

Reply:
Can you confirm if this is still an issue following the update and if so which version of TypeScript?

Reply:
I'll double-check it on the weekend

Reply:
When I enabled tests in another branch (which is created from `master` branch), I could reproduce the problem: commit https://github.com/ts-essentials/ts-essentials/pull/413/commits/64fffdc70bf9a5e86482fadb894c8604e7baa92b

<img width="668" alt="Screenshot 2024-10-07 at 09 31 25" src="https://github.com/user-attachments/assets/ec360a94-a772-4681-9de2-e1c3781af4b5">

The problem existed in TypeScript versions 4.5-4.8, it's fixed after this commit - https://github.com/ts-essentials/ts-essentials/pull/413/commits/98258d5fbccd4e6d0ba4d3e32ec58ce42c9f33ab

I'll get it merged today, please rebase it and you won't need to make any other changes